backend/src/game/newcards/discard_cards/victory_point_manipulation_cards/test_apartment_building.py

```diff
@@ -0,0 +1,29 @@
+import pytest
+
+from backend.src.game.newcards.discard_cards.victory_point_manipulation_cards.apartment_building import \
+    ApartmentBuilding
+from backend.src.game.player import Player
+
+
+@pytest.fixture(autouse=True)
+def player():
+    player = Player()
+    return player
+
+
+def test_apartment_building_adds_3_victory_points(player):
+    ApartmentBuilding().immediate_effect(player, None)
+    assert player.victory_points == 3
+
+
+def test_apartment_building_costs_5_energy():
+    assert ApartmentBuilding().cost == 5
+
+
+def test_apartment_building_has_effect():
+    effect = ApartmentBuilding().effect
+    assert effect and not effect.isspace()
```

Comment:
@gpfaia Is there a better way to write this?

Reply:
This is similar to my comment about footnote.  If it is determined that there are elements that will not have effects, then you should use the None in the __init__.  If it will always have a value then it makes sense to write just this since that is what you want to test, you are not writing a generic test for this one.

`assert effect == "+ 3[Star]"`

In the future if writing a more generic
https://stackoverflow.com/a/45968236

Reply:
I don't want to assert on the string value of the effect. If we go down that path, it'll be painful. Pull Requests are there to make sure the strings for the cards are correct. Duplicating that in the test doesn't add value.
Good to know about the =None, I will do that.

Reply:
If that is the case, you can probably get rid of the footnote test case then as it won't be needed.

For this test,  I think it would be safe to  just adding an assert type str for effect into the NewCard class's init method.  The secondary condition is not necessary as you're basically repeating code.

I think what you may have been looking for is
`assert effect and isinstance(effect, str)`

Reply:
> Is this a replacement for the other cards folder that already exists because some of the code here is repetitive?

Yes